tests/performance/profiling.py

```diff
@@ -0,0 +1,77 @@
+import timeit
+import tracemalloc
+from multiprocessing import Process, Value
+
+import pandas as pd
+
+
+def _profile_time(method, dataset, iterations=100):
+    total_time = 0
+    for _ in range(iterations):
+        start_time = timeit.default_timer()
+        method(dataset)
+        total_time += timeit.default_timer() - start_time
+
+    return total_time / iterations
+
+
+def _set_memory_for_method(method, dataset, peak_memory):
+    tracemalloc.start()
+    method(dataset)
+    peak_memory.value = tracemalloc.get_traced_memory()[1]
+    tracemalloc.stop()
+
+
+def _profile_memory(method, dataset):
+    peak_memory = Value('i', 0)
+    profiling_process = Process(target=_set_memory_for_method, args=(method, dataset, peak_memory))
+    profiling_process.start()
+    profiling_process.join()
+    return peak_memory.value
+
+
+def profile_transformer(transformer, dataset_generator, transform_size, fit_size=None):
+    """
+    Function to profile a transformer.
+
+    This function will get the total time and peak memory
+    for the ``fit``, ``transform`` and ``reverse_transform``
+    methods of the provided transformer against the provided
+    dataset.
+
+    Args:
+        transformer (Transformer):
+            Transformer instance.
+        dataset_generator (DatsetGenerator):
+            DatasetGenerator instance.
+        transform_size (int):
+            Number of rows to generate for ``transform`` and ``reverse_transform``.
+        fit_size (int or None):
+            Number of rows to generate for ``fit``. If None, use ``transform_size``.
+    """
+    transformer_instance = transformer()
+    fit_size = fit_size or transform_size
+    fit_dataset = dataset_generator.generate(fit_size)
+    fit_time = _profile_time(transformer_instance.fit, fit_dataset)
+    fit_memory = _profile_memory(transformer_instance.fit, fit_dataset)
+
+    transform_dataset = dataset_generator.generate(transform_size)
+    transform_time = _profile_time(transformer_instance.transform, transform_dataset)
```

Comment:
same as above wrt to running twice, but here the deepcopy may not be necessary, since transform and reverse_transform should not alter the internal state of the transformer.

Reply:
what do you mean by run twice?